.clang-format

```diff
@@ -0,0 +1,192 @@
+---
```

Comment:
This is new. How does it compare to the conventions written down by Jeroen somewhere in the OpenQL environment?

Reply:
Hi Hans, I tried to use the most similar conventions as OpenQL. For reference, this .clang-format file is generated like this:
`clang-format -style=llvm -dump-config > .clang-format`

The good thing with automated formatting is that the parts of https://openql.readthedocs.io/en/latest/developer/conventions.html that concern code style no longer need to exist, since clang-format would do it automatically.

What we can add is a git hook that automatically formats code whenever you commit.
OpenQL's code is well formatted, but the simulator's code was horrible, hence these changes to automatically format everything.